### PR TITLE
Add support for LoongArch64

### PR DESCRIPTION
**Pull request by heiher**
Add support for [LoongArch64](https://doc.rust-lang.org/rustc/platform-support/loongarch-linux.html) (A tier-2 with host tools target).

### CODE REVIEW

**Comment by heiher** [comment]
### Test resutls

PASSED

```bash
cargo test --features instrumentation
```

```
running 36 tests
test compress::tests::test_compression_from_str ... ok
test db::tests::test_db_open_should_fail ... ok
test compress::tests::test_compression_interfaces ... ok
test index::test::test_entries ... ok
test index::test::test_find_entries ... ok
test index::test::test_find_entry_same_value ... ok
test index::test::test_find_entry_zero_pk ... ok
test index::test::test_find_any_entry ... ok
test table::test::oversized_into_fixed_panics - should panic ... ok
test table::test::bad_size_header ... ok
test table::test::insert_simple ... ok
test db::tests::test_partial_log_recovery ... ok
test table::test::ref_underflow ... ok
test table::test::remove_simple ... ok
test table::test::replace_multipart_shorter ... ok
test table::test::ref_counting ... ok
test db::tests::test_recover_from_log_on_error ... ok
test table::test::replace_simple ... ok
test table::test::replace_multipart_longer ... ok
test db::tests::test_db_open_fail_then_recursively_create ... ok
test db::tests::test_db_open_or_create ... ok
test table::test::multipart_collision ... ok
test migration::test::clear_column ... ok
test db::tests::test_add_column ... ok
test db::tests::test_indexed_keyvalues ... ok
test db::tests::test_remove_column ... ok
test db::tests::test_indexed_overlay_against_backend ... ok
test db::tests::test_indexed_btree_1 ... ok
test table::test::iteration ... ok
test db::tests::test_indexed_btree_2 ... ok
test migration::test::migrate_simple ... ok
test db::tests::test_continue_reindex ... ok
test db::tests::test_indexed_btree_3 ... ok
test db::tests::test_simple ... ok
test db::tests::test_btree_iter ... ok
test db::tests::test_random ... ok

test result: ok. 36 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 10.03s

     Running tests/loom.rs (target/debug/deps/loom-3e812a113b89ec98)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests parity-db

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

**Comment by arkpar** [comment]
Thank you for the PR. 
We can't support it officially since there's not CI for this platform. There's a chance that something may break in future releases and the maintainers have no way to debug and test fixes.
It should be possible to setup a qemu-based CI action for running tests at least, but I could not find any readily available solution for this.

File: src/lib.rs [hunk 1]
```diff
@@ -33,8 +33,8 @@ pub use stats::{ColumnStatSummary, StatSummary};
 pub const KEY_SIZE: usize = 32;
 pub type Key = [u8; KEY_SIZE];
 
-#[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
-compile_error!("parity-db only supports x86_64 and aarch64");
+#[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "loongarch64")))]
+compile_error!("parity-db only supports x86_64, aarch64 and loongarch64");
```
Comment (arkpar) [inline src/lib.rs:37]:
```suggestion
compile_error!("parity-db only supports x86_64, aarch64 and loongarch64 (unofficially)");
```

**Review by arkpar — CHANGES_REQUESTED** [review]

**Comment by heiher** [comment]
Thanks for your review. I'll add CI when it's ready.

**Review by arkpar — APPROVED** [review]

**Pull request merged**